samplesheets/views_api.py

```diff
@@ -310,6 +316,182 @@ def post(self, request, *args, **kwargs):
+
+    **URL:** ``/samplesheets/api/irods/ticket/retrieve/{IrodsAccessTicket.sodar_uuid}``
+
+    **Methods:** ``GET``
+    """
+
+    lookup_field = 'sodar_uuid'
+    lookup_url_kwarg = 'irodsaccessticket'
+    permission_required = 'samplesheets.edit_sheet'
+    serializer_class = IrodsAccessTicketSerializer
+
+    def get_object(self):
+        """Override get_object() to get IrodsAccessTicket"""
+        return IrodsAccessTicket.objects.get(
+            sodar_uuid=self.kwargs.get('irodsaccessticket')
+        )
+
+
+class IrodsAccessTicketCreateAPIView(
+    IrodsAccessTicketModifyMixin, SODARAPIGenericProjectMixin, CreateAPIView
+):
+    """
+    Create an iRODS access ticket for a project.
+
+    **URL:** ``/samplesheets/api/irods/ticket/create/{Project.sodar_uuid}``
+
+    **Methods:** ``POST``
+
+    **Parameters:**
+
+    - ``path``: iRODS path
+    - ``label``: Label (string, optional)
+    - ``date_expires``: Expiration date (YYYY-MM-DDThh:mm:ssZ, optional)
+    """
+
+    permission_required = 'samplesheets.edit_sheet'
+    serializer_class = IrodsAccessTicketSerializer
+
+    def get_serializer_context(self):
+        context = super().get_serializer_context()
+        context['project'] = self.get_project()
+        context['user'] = self.request.user
+        return context
+
+    def perform_create(self, serializer):
+        """Override perform_create() to create IrodsAccessTicket"""
+        irods_backend = get_backend_api('omics_irods')
+        try:
+            with irods_backend.get_session() as irods:
+                ticket = irods_backend.issue_ticket(
+                    irods,
+                    'read',
+                    serializer.validated_data.get('path'),
+                    ticket_str=build_secret(16),
+                    expiry_date=serializer.validated_data.get('date_expires'),
+                )
+        except Exception as ex:
+            raise ValidationError(
+                '{} {}'.format('Creating ' + IRODS_TICKET_EX_MSG + ':', ex)
+            )
+
+        serializer.validated_data['ticket'] = ticket.ticket
+        serializer.save()
+        # Create timeline event
+        self.create_timeline_event(serializer.instance, 'create')
+        # Add app alerts to owners/delegates
+        self.create_app_alerts(serializer.instance, 'create', self.request.user)
+
+
+class IrodsAccessTicketUpdateAPIView(
+    IrodsAccessTicketModifyMixin, SODARAPIGenericProjectMixin, UpdateAPIView
+):
+    """
+    Update an iRODS access ticket for a project.
+
+    **URL:** ``/samplesheets/api/irods/ticket/update/{IrodsAccessTicket.sodar_uuid}``
+
+    **Methods:** ``PUT``, ``PATCH``
+
+    **Parameters:**
+
+    - ``label``: Label (string)
+    - ``date_expires``: Expiration date (YYYY-MM-DD)
+    """
+
+    lookup_url_kwarg = 'irodsaccessticket'
+    permission_required = 'samplesheets.edit_sheet'
+    serializer_class = IrodsAccessTicketSerializer
+
+    def get_object(self):
```

Comment:
Same thing here, the `get_object()` override should be replaceable with `queryset_project_field`.